tracer/src/Datadog.Trace/Sampling/GlobMatcher.cs

```diff
@@ -0,0 +1,34 @@
+// <copyright file="GlobMatcher.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+
+#if NETCOREAPP3_1_OR_GREATER
+using Datadog.Trace.Vendors.IndieSystem.Text.RegularExpressions;
+#else
+using System.Text.RegularExpressions;
+#endif
+
+#nullable enable
+
+namespace Datadog.Trace.Sampling;
+
+internal static class GlobMatcher
+{
+    public static Regex BuildRegex(string glob, RegexOptions? regexOptions = null, TimeSpan? regexTimeout = null)
+    {
+        // TODO default glob (maybe null/empty/whitespace) should be *
```

Comment:
I'm reading this TODO again after a long time and I think I may need to look at the single span sampling RFC again as I think the glob patterns were supposed to be defaulted to `*`, but it looks like they don't allow for null/empty/whitespace. Not something that this PR would have to do.

Wondering if it'd be best to remove this TODO though (unless it applies to the custom sampling rules too)?

Reply:
Thanks. I think we already handle `null` by just matching everything (this is the equivalent of `"*"`), but I can handle this as part of these PRs, if we also need to handle empty/whitespace strings.